src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -1146,7 +1146,7 @@ public Map<String, Boolean> mapTenants(final User user, Set<String> roles) {
                         // Indeed, because we don't have control over what will be
                         // passed on as values of users' attributes, we have to make
                         // sure that we don't allow them to select tenants that do not exist.
-                        if (ConfigModelV7.this.tenants.getCEntries().keySet().contains(tenant)) {
```

Comment:
Actually the call to `getCEntries()` is fine, but it should be done outside of this loop so that its only requested once per request and not per tenant for every tenant resolvable to a user's mapped roles